latch/registry/table.py

```diff
@@ -690,20 +698,90 @@ def _add_column_upserts_selection(
 
         args = l.ArgumentNode()
         args.name = _name_node("input")
-        args.value = _json_value(
-            {
-                "argExperimentId": self.table.id,
-                "argKeys": keys,
-                "argTypes": _var_node(argTypesVar),
-            }
-        )
+        args.value = _json_value({
+            "argExperimentId": self.table.id,
+            "argKeys": keys,
+            "argTypes": _var_node(argTypesVar),
+        })
 
         res.alias = _name_node(f"upd{len(mutations)}")
         res.arguments = tuple([args])
 
         mutations.append(res)
         vars[argTypesVar] = (l.parse_type("[JSON]!"), types)
 
+    def _get_unresolved_blobs_in_db_val(
```

Comment:
this and `_get_unresolved_blobs_in_update` should be global utility functions, not members of `Table`